# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/gradient_shared.glsl
# Commit: 8d13c0ab66f0
# Full Hash: 8d13c0ab66f0ac137b1461d3451ecb9361ec2d5e
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2023-03-22 21:15:54
# Regressor Bug: 1823411
# File Overlap Count: 1
# Description:
#   Bug 1823411 - Give all webrender's shader varyings an explicit precision. r=gfx-reviewers,lsalzman
#   
#   Mali profiling tools have shown we are frequently fragment bound, in
#   particular due to varying interpolation. To help mitigate this, we
#   should use mediump where possible. Currently most of our varyings are
# ==============================================================================

diff -r 3919fbd972a6 -r 8d13c0ab66f0 gfx/wr/webrender/res/gradient_shared.glsl
--- a/gfx/wr/webrender/res/gradient_shared.glsl	Tue Mar 21 19:04:22 2023 +0000
+++ b/gfx/wr/webrender/res/gradient_shared.glsl	Tue Mar 21 19:05:19 2023 +0000
@@ -7,12 +7,12 @@
 // Size of the gradient pattern's rectangle, used to compute horizontal and vertical
 // repetitions. Not to be confused with another kind of repetition of the pattern
 // which happens along the gradient stops.
-flat varying vec2 v_repeated_size;
+flat varying highp vec2 v_repeated_size;
 
-varying vec2 v_pos;
+varying highp vec2 v_pos;
 
 #ifdef WR_FEATURE_ALPHA_PASS
-flat varying vec2 v_tile_repeat;
+flat varying highp vec2 v_tile_repeat;
 #endif
 
 #ifdef WR_VERTEX_SHADER